src/AccessControlHierarchyBase.php

```diff
@@ -168,16 +168,65 @@ public function load($id) {
     }
   }
 
+  /**
+   * Check to see if this entity is acting as Access Control.
+   *
+   * Any entity that is not being used for Access Control needs to return
+   * FALSE.
+   *
+   * @param EntityInterface $entity
+   *   An entity to check to see if it is acting as Access Control.
+   *
+   * @return bool
+   *   TRUE if Access Control entity, FALSE otherwise.
+   */
+  protected abstract function isAccessControlEntity(EntityInterface $entity);
```

Comment:
we can't add this in a beta release, its an API change, would require a new major version. I think we should avoid it.

Reply:
@larowlan -- So, according to "[What are alpha and beta releases, and release candidates?](https://www.drupal.org/docs/8/understanding-drupal-version-numbers/what-are-alpha-and-beta-releases-and-release-candidates)", it is permissible to make an API change in the beta version. I agree that API changes should be avoided in beta versions, but I think it would be warranted in this case because the lack of a check on the controlling entity can cause unanticipated side effects, *especially* when you are integrating Workbench Access with third-party RBAC systems.

Reply:
I don't consider adding functionality to be an API change. This is an addition, not a diff, so it won't break any existing calls.

Reply:
This is now:

```
 protected function isAccessControlEntity(EntityInterface $entity) {
    return FALSE;
  }
```